4-float-features/4-float-features.md

```diff
@@ -0,0 +1,74 @@
+# Introduction
+
+Num type has been separated into Int type and Float type.\
+The old Num type operators and features only support Int types, so there is a need to support Float types also.
+
+We propose the addition of basic operators to Float types to match those of Int types.\
+Also, we propose more support in recognizing Int and Float types, such as operator type prediction.\
+When adding these features, consistency will be the key.
+
+# Float Operators
+*overloading will be avoided until a more general consensus for overloading is set.
+
+## Boolean Float Operators
+
+There should be boolean operators for float types too.\
+Currently, there are 3 boolean operators, only supporting Int types.
+* `<`
+* `>`
+* `=`
+
+We propose the addition of the same operators, but for Float types.\
+In order to be consistent with Float Ops for plus, minus, and times, we will **Not** overload the operators*.\
+Instead, the operators will have a following **"."** after them.
+* `<.`
+* `>.`
+* `=.`
+
+## division operator
+
+There should be a division operator to complete the set of basic arithmetic operators.\
+There should be one operator `/` for Integer division and another for float division `/.` to avoid overloading*.\
+Also, casting will be avoided for now.
+
+Integer division `/` should take two integer types and return the floor of the result.
+    ```
+    Ex.
+    2 / 2 -> 1
+    5 / 2 -> 2
+    ``` 
+
+Float division `/.` should take two float types and return a float type based on the result.
+    ```
+    Ex.
+    2. /. 2. -> 1.
+    5. /. 2. -> 2.5
+    '''
+Float division could lead to edge cases such as:
+    '''
+    1.0 /. 0.0
+    -1.0 /. 0.0
+    0.0 /. 0.0
+    '''
+Due to this, we suggest introducing three new constants, following the IEEE 754 format like Ocaml:
+* `NaN`
+* `Inf`
+* `Neg_Inf`
```

Comment:
I think we don't need Neg_Inf once we have unary negation, but what should the syntax be to distinguish integer negation from float negation?

Reply:
I think it would be more confusing to introduce a separate syntax for negation of int/float. I was considering to simply allow the construction of "-" char in a [Int|Float]Lit such that we could have negative numbers. Is there a reason that I missed for why we should have a separate syntax for the two types?

Reply:
unary negation on literals would work fine, but negation works on any expression in most languages, e.g. `-x`

Reply:
if we have unary negation on literals only then we can do `-1. *. x`, but that's a little weird

Reply:
we could make it so the type synthesized by `-e` depends on the type synthesized by `e` rather than making `e` analytic, as it is in binary ops. that's probably the cleanest option.

Reply:
anyway unary negation is worth its own PHI honestly, its tricky. don't worry about negation for this one.

Reply:
It might make more sense if I took out the section about negation from Int/Floats PHI and create a separate PHI for negation of any literal in Hazel.